axelrod/strategies/punisher.py

```diff
@@ -119,3 +119,42 @@ def reset(self):
         self.grudged = False
         self.grudge_memory = 0
         self.mem_length = 1
+
+class LevelPunisher(Player):
+    """
+    Names:
+
+    - Level Punisher: Original by Geraint Palmer
+    """
+
+    name = 'Level Punisher'
+    classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        super(LevelPunisher, self).__init__()
+        self.history = []
+        self.mem_length = 1
+        self.grudged = False
+        self.grudge_memory = 1
+        
+    def strategy(self, opponent):
+        
+        if self.grudge_memory >= self.mem_length:
+            self.grudge_memory = 0
+            self.grudged = False
```

Comment:
@souravsingh could you give us some more details about what this strategy is supposed to do please? Perhaps link to the description from CoopSim?